error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ClassMemberOrdering.java

```diff
@@ -0,0 +1,181 @@
+  }
+
+  private static SuggestedFix replaceClassMembers(
+      ImmutableList<ClassMemberWithComments> classMembers,
+      ImmutableList<ClassMemberWithComments> replacementClassMembers,
+      VisitorState state) {
+    return Streams.zip(
+            classMembers.stream(),
+            replacementClassMembers.stream(),
+            (original, replacement) -> replaceClassMember(state, original, replacement))
+        .reduce(SuggestedFix.builder(), SuggestedFix.Builder::merge, SuggestedFix.Builder::merge)
+        .build();
+  }
+
+  private static SuggestedFix replaceClassMember(
+      VisitorState state, ClassMemberWithComments original, ClassMemberWithComments replacement) {
+    /* Technically this check is not necessary, but it avoids redundant replacements. */
+    if (original.equals(replacement)) {
+      return SuggestedFix.emptyFix();
+    }
+
+    String replacementSource =
+        Stream.concat(
+                replacement.comments().stream(),
+                Stream.of(SourceCode.treeToString(replacement.tree(), state)))
+            .collect(joining(System.lineSeparator()));
+    return SuggestedFixes.replaceIncludingComments(
+        TreePath.getPath(state.getPath(), original.tree()), replacementSource, state);
+  }
+
+  /** Returns the class' members with their comments. */
+  private static ImmutableList<ClassMemberWithComments> getClassMembersWithComments(
+      ClassTree classTree, VisitorState state) {
+    return classTree.getMembers().stream()
+        .map(
+            member ->
+                new AutoValue_ClassMemberOrdering_ClassMemberWithComments(
+                    member, getClassMemberComments(state, classTree, member)))
+        .collect(toImmutableList());
+  }
+
+  private static ImmutableList<String> getClassMemberComments(
+      VisitorState state, ClassTree classTree, Tree classMember) {
+    int typeStart = ASTHelpers.getStartPosition(classTree);
+    int typeEnd = state.getEndPosition(classTree);
+    int memberStart = ASTHelpers.getStartPosition(classMember);
+    int memberEnd = state.getEndPosition(classMember);
+    if (typeStart == Position.NOPOS
+        || typeEnd == Position.NOPOS
+        || memberStart == Position.NOPOS
+        || memberEnd == Position.NOPOS) {
+      /* Source code details appear to be unavailable. */
+      return ImmutableList.of();
+    }
+
+    Optional<Integer> previousClassTokenEndPos =
+        state.getOffsetTokens(typeStart, typeEnd).stream()
+            .map(ErrorProneToken::endPos)
+            .takeWhile(endPos -> endPos < memberStart)
+            .reduce((earlierPos, laterPos) -> laterPos);
+
+    List<ErrorProneToken> classMemberTokens =
+        state.getOffsetTokens(previousClassTokenEndPos.orElse(memberStart), memberEnd);
+
+    return classMemberTokens.get(0).comments().stream()
+        .map(Tokens.Comment::getText)
+        .collect(toImmutableList());
+  }
+
+  @AutoValue
+  abstract static class ClassMemberWithComments {
```

Comment:
Considering the `Class -> Type` rename we're doing, wdyt of `TypeMember` or just `Member`?

Asking because `TypeMember` sounded weird to me initially. Probably I'm just not used to it.

Reply:
Yeah, `TypeMember` can work! (Doesn't sound so weird to me :smile:.)